java-squid/src/main/java/org/sonar/java/model/expression/MemberSelectExpressionTreeImpl.java

```diff
@@ -67,18 +69,17 @@ public MemberSelectExpressionTreeImpl(ExpressionTree expression, IdentifierTree
   public MemberSelectExpressionTreeImpl(AstNode astNode, ExpressionTree expression, IdentifierTree identifier) {
     super(astNode);
 
-    this.dims = -1;
+    this.dims = ImmutableList.<AnnotatedDimension>of();
     this.expression = Preconditions.checkNotNull(expression);
     this.identifier = Preconditions.checkNotNull(identifier);
   }
 
   public MemberSelectExpressionTreeImpl completeWithExpression(ExpressionTree expression) {
-    Preconditions.checkState(dims >= 0 && this.expression == null);
+    Preconditions.checkState(dims.size() >= 0 && this.expression == null);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Bad comparison of nonnegative value with 0 in org.sonar.java.model.expression.MemberSelectExpressionTreeImpl.completeWithExpression(ExpressionTree) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AINT_BAD_COMPARISON_WITH_NONNEGATIVE_VALUE)